packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -35,14 +35,26 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
     uint256 public liquidReserves;
 
     // Reserves currently utilized due to L2-L1 transactions in flight.
-    uint256 public utilizedReserves;
+    int256 public utilizedReserves;
+
+    // Exponential decay exchange rate to accumulate fees to LPs over time.
+    uint256 public lpFeeRatePerSecond;
+
+    // Last timestamp that LP fees were updated.
+    uint256 public lastLpFeeUpdate;
+
+    // Cumulative undistributed LP fees. As fees accumulate, they are subtracted from this number.
+    uint256 public undistributedLpFees;
+
+    // How long, in seconds, it takes for transfers to move from L2 optimism to L1 Ethereum.
+    uint256 public optimismL2toL1Time = 1 weeks;
 
     // Administrative contract that deployed this contract and also houses all state variables needed to relay deposits.
     BridgeAdminInterface public bridgeAdmin;
 
     // A Relay represents a an attempt to finalize a cross-chain transfer that originated on an L2 DepositBox contract
     // and can be bridged via this contract.
-    enum RelayState { Uninitialized, Pending, Finalized }
+    enum RelayState { Uninitialized, Pending, Disputed, Finalized }
```

Comment:
Instead of adding a new `mapping` to keep track of disputed deposits, I added another state to this `enum` to make it more readable and clear about which deposits have been disputed. The downside to this change is that an additional condition has to be added to `relayDeposit()` about what `RelayState` relays can be in to be relayed.